### PR TITLE
[PHEE-697]Adding payee dfpsid header to transfer API

### PR DESCRIPTION
**Pull request by apurbraj**
Sending payeedfspid varibale to workflow

Pending Task - Validating payee dfspId [PHEE-698](https://fynarfin.atlassian.net/browse/PHEE-698)
## Description

* PR title should have jira ticket enclosed in `[]`.<br>
  Format: ``` [jira_ticket] description```<br>
  ex: [phee-123] PR title.
* Add a link to the Jira ticket.
* Describe the changes made and why they were made.

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!
- [ ] Followed the PR title naming convention mentioned above.

- [ ] Design related bullet points or design document link related to this PR added in the description above.

- [ ] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Add required Swagger annotation and update API documentation with details of any API changes if applicable

- [ ] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

[PHEE-698]: https://fynarfin.atlassian.net/browse/PHEE-698?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/org/mifos/connector/channel/api/definition/TransferApi.java [hunk 1]
```diff
@@ -21,6 +22,7 @@ GsmaP2PResponseDto transfer(@RequestHeader(value = "Platform-TenantId") String t
             @RequestHeader(value = BATCH_ID_HEADER, required = false) String batchId,
             @RequestHeader(value = CLIENTCORRELATIONID, required = false) String correlationId,
             @RequestHeader(value = REGISTERING_INSTITUTION_ID, required = false) String registeringInstitutionId,
+            @RequestHeader(value = PAYEE_DFSP_ID, required = false) String payeeDFSPId,
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/channel/api/definition/TransferApi.java:25]: create an issue report for this validation

File: src/main/java/org/mifos/connector/channel/api/implementation/TransferApiController.java [hunk 2]
```diff
@@ -34,13 +35,15 @@ public class TransferApiController implements TransferApi {
...
+            String payeeDFSPId, TransactionChannelRequestDTO requestBody) throws JsonProcessingException {
         Headers headers = new Headers.HeaderBuilder().addHeader("Platform-TenantId", tenant).addHeader(BATCH_ID, batchId)
-                .addHeader(CLIENTCORRELATIONID, correlationId).addHeader(REGISTERING_INSTITUTION_ID, registeringInstitutionId).build();
+                .addHeader(CLIENTCORRELATIONID, correlationId).addHeader(REGISTERING_INSTITUTION_ID, registeringInstitutionId)
+                .addHeader(PAYEE_DFSP_ID, payeeDFSPId).build();
         Exchange exchange = SpringWrapperUtil.getDefaultWrappedExchange(producerTemplate.getCamelContext(), headers,
                 objectMapper.writeValueAsString(requestBody));
         logger.info("Client correlation id: " + correlationId);
         logger.info("Batch id: " + batchId);
+        logger.info("Payee DFSP Id: " + payeeDFSPId);
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/channel/api/implementation/TransferApiController.java:46]: donot use + for logger contatenation, use {}

File: src/main/java/org/mifos/connector/channel/camel/config/CamelProperties.java [hunk 3]
```diff
@@ -13,5 +13,5 @@ private CamelProperties() {}
     public static final String PAYMENT_SCHEME_HEADER = "X-Payment-Scheme";
     public static final String REGISTERING_INSTITUTION_ID = "X-Registering-Institution-ID";
     public static final String PARTY_LOOKUP_FSP_ID = "partyLookupFspId";
-
+    public static final String PAYEE_DFSP_ID = "X-PayeeDFSP-ID";
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/channel/camel/config/CamelProperties.java:16]: use this naming convention only for API header. don't use it for Zeebe variable

File: src/main/java/org/mifos/connector/channel/api/implementation/TransferApiController.java (hunk 2 above)
Comment (fynmanoj) [reply src/main/java/org/mifos/connector/channel/api/implementation/TransferApiController.java:46]: mark this log to debug level.

File: src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java [hunk 4]
```diff
@@ -319,14 +322,16 @@ private void transferRoutes() {
                     }
                     extraVariables.put(TENANT_ID, tenantId);
                     extraVariables.put(REGISTERING_INSTITUTION_ID, registeringInstitutionId);
+                    extraVariables.put(PAYEE_DFSP_ID, payeeDFSPId);
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java:325]: use this naming convention only for API header. don't use it for Zeebe variable

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java (hunk 4 above)
Comment (apurbraj) [reply src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java]: changed the variable to payeeDFSPId

**Review by fynmanoj — APPROVED** [review]

**Pull request closed without merging**